Modules/_io/winconsoleio.c

```diff
@@ -11,7 +11,7 @@
 #include "pycore_fileutils.h"     // _Py_BEGIN_SUPPRESS_IPH
 #include "pycore_object.h"        // _PyObject_GC_UNTRACK()
 
-#ifdef MS_WINDOWS
+#if defined(MS_WINDOWS) && !defined(MS_WINDOWS_NON_DESKTOP)
```

Comment:
I'd only skip defining the `io._WindowsConsoleIO` type for `MS_WINDOWS_GAMES` builds.

Python 3.12 requires Windows 10+, which supports [UWP console apps](https://learn.microsoft.com/en-us/windows/uwp/launch-resume/console-uwp). In the Windows 11 SDK, most of the console API, including `ReadConsoleW()` and `WriteConsoleW()`, supports `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_APP | WINAPI_PARTITION_SYSTEM)`. The app API partition includes desktop apps, store apps, and phone apps.